README.md

```diff
@@ -1,57 +1,297 @@
+                nic_conf.subnet = subnet
+                nic_conf.public_ipaddress = public_ip
+            end
+        ]
+    end
+)
+```
+
+<a id="vm"></a>
+### Create a virtual machine with system identity
+
+```ruby
+puts 'Creating a Ubuntu 16.04.0-LTS Standard DS2 V2 virtual machine w/ a public IP'
+vm_create_params = ComputeModels::VirtualMachine.new.tap do |vm|
+  vm.location = location
+  vm.os_profile = ComputeModels::OSProfile.new.tap do |os_profile|
+    os_profile.computer_name = vm_name
+    os_profile.admin_username = 'notAdmin'
+    os_profile.admin_password = 'Pa$$w0rd92'
+  end
+
+  vm.storage_profile = ComputeModels::StorageProfile.new.tap do |store_profile|
+    store_profile.image_reference = ComputeModels::ImageReference.new.tap do |ref|
+      ref.publisher = 'canonical'
+      ref.offer = 'UbuntuServer'
+      ref.sku = '16.04.0-LTS'
+      ref.version = 'latest'
+    end
+    store_profile.os_disk = ComputeModels::OSDisk.new.tap do |os_disk|
+      os_disk.name = "sample-os-disk-#{vm_name}"
+      os_disk.caching = ComputeModels::CachingTypes::None
+      os_disk.create_option = ComputeModels::DiskCreateOptionTypes::FromImage
+      os_disk.vhd = ComputeModels::VirtualHardDisk.new.tap do |vhd|
+        vhd.uri = "https://#{storage_acct.name}.blob.core.windows.net/rubycontainer/#{vm_name}.vhd"
+      end
+    end
+  end
+
+  vm.hardware_profile = ComputeModels::HardwareProfile.new.tap do |hardware|
+    hardware.vm_size = ComputeModels::VirtualMachineSizeTypes::StandardDS2V2
+  end
+
+  vm.network_profile = ComputeModels::NetworkProfile.new.tap do |net_profile|
+    net_profile.network_interfaces = [
+        ComputeModels::NetworkInterfaceReference.new.tap do |ref|
+          ref.id = nic.id
+          ref.primary = true
+        end
+    ]
+  end
+
+  # Use System Assigned Identity for the VM
+  vm.identity = ComputeModels::VirtualMachineIdentity.new.tap do |identity|
+    identity.type = ComputeModels::ResourceIdentityType::SystemAssigned
+  end
+end
+
+ssh_pub_location = File.expand_path('~/.ssh/id_rsa.pub')
+if File.exists? ssh_pub_location
+  puts "Found SSH public key in #{ssh_pub_location}. Disabling password and enabling SSH authentication."
+  key_data = File.read(ssh_pub_location)
+  puts "Using public key: #{key_data}"
+  vm_create_params.os_profile.linux_configuration = ComputeModels::LinuxConfiguration.new.tap do |linux|
+    linux.disable_password_authentication = true
+    linux.ssh = ComputeModels::SshConfiguration.new.tap do |ssh_config|
+      ssh_config.public_keys = [
+          ComputeModels::SshPublicKey.new.tap do |pub_key|
+            pub_key.key_data = key_data
+            pub_key.path = '/home/notAdmin/.ssh/authorized_keys'
+          end
+      ]
+    end
+  end
+end
+
+print_item vm = compute_client.virtual_machines.create_or_update(GROUP_NAME, "sample-ruby-vm-#{vm_name}", vm_create_params)
+```
+
+<a id="extension"></a>
+### Add MSI extension to the VM
```

Comment:
Is there more we could explain here? are there some MSI docs we can refer to from Azure?
anything on where the values shown below should be coming from?

Reply:
sure